cranelift/codegen/src/machinst/helpers.rs

```diff
@@ -0,0 +1,35 @@
+//! Miscellaneous helpers for machine backends.
+
+use crate::ir::types::*;
+use crate::ir::Type;
+
+/// Returns the size (in bits) of a given type.
+pub fn ty_bits(ty: Type) -> usize {
+    match ty {
+        B1 => 1,
+        B8 | I8 => 8,
+        B16 | I16 => 16,
+        B32 | I32 | F32 | R32 => 32,
+        B64 | I64 | F64 | R64 => 64,
+        B128 | I128 => 128,
+        IFLAGS | FFLAGS => 32,
+        B8X8 | I8X8 | B16X4 | I16X4 | B32X2 | I32X2 => 64,
+        B8X16 | I8X16 | B16X8 | I16X8 | B32X4 | I32X4 | B64X2 | I64X2 => 128,
+        F32X4 | F64X2 => 128,
+        _ => panic!("ty_bits() on unknown type: {:?}", ty),
+    }
+}
+
+pub(crate) fn ty_is_int(ty: Type) -> bool {
```

Comment:
Could this function be renamed to something that exposes the reason for the query?  I mean, `ty_is_int` is, well, why do we care?  If it were (eg) `param_of_this_ty_goes_in_intreg` then that would be clearer.  But that's a decision that's surely per-target?  Why would target-independent code want to know about the apparent int-ness of a type?

Reply:
Removed, actually, as we can use other existing helpers -- thanks!